src/BuiltinExtensions/ImageBatchTool/ImageBatchToolExtension.cs

```diff
@@ -18,6 +18,18 @@ public class ImageBatchToolExtension : Extension
 {
     public static PermInfo PermUseImageBatchTool = Permissions.Register(new("imagebatcher_use_image_batcher", "[Image Batch Tool] Use Image Batcher", "If true, the user may use the Image Batcher tool. Only makes sense for localhost users.", PermissionDefault.ADMINS, Permissions.GroupUser));
 
+    //   the following two statics are used to handle when one input file
+    //   produces multiple output images.In that case the first image is output
+    //   as normal "output_dir/preExt.ext" but any images after that will be of the
+    //   form "output_dir/preExt_lastFileCount.ext" 
+
+    // the preext of the last file written out
+    public static string lastFilePreExt;
```

Comment:
this should not be static at all 0.o

Reply:
Oops, sorry, didn't see the for loop scope it was inside of. Moved to that scope. Hopefully that is the right approach. Seems to work in my local tests. If you want the commits squashed etc let me know. Bit rusty.